LFS-1206: Questionnaire/Subject selector does not reset internal state when closed before completion

IntegralProgrammer opened the pull request:
This PR causes the selected Subject and selected Questionnaire within the _New Form_ dialog to be reset to empty if the _New Form_ dialog is closed without actually creating a new Form.

To test, follow the instructions on https://phenotips.atlassian.net/browse/LFS-1206 on the `dev` branch to reproduce the bug. After that, build and run the `LFS-1206` branch and follow the same instructions as before this time ensuring that by _step 8_ the _New Form_ dialog is reset to its initial state and there are no initially selected _Subjects_ or _Questionnaires_.

IntegralProgrammer committed:
LFS-1206: Questionnaire/Subject selector does not reset internal state when closed before completion

IntegralProgrammer committed:
LFS-1206: Questionnaire/Subject selector does not reset internal state when closed before completion

Additionally, reset the selectedQuestionnaire state variable

sashaandjic commented:
Now I can create 2 Patient Information forms for the same patient.
I followed the steps. At the end I selected John even though the selection option was grayed out (but was not disabled). 
I think I saw this earlier, but was not able to reproduce it.
![image](https://user-images.githubusercontent.com/23243244/122613752-5e6d0c80-d053-11eb-8f8e-77f6d8cbb4d9.png)

sashaandjic commented:
Bah, I now I can no longer reproduce it again!

sashaandjic reviewed (approved):
tested

sdumitriu reviewed (changes requested):
Still not entirely fixed:

- lfs mode
- create a new Patient Information form for a new patient, save, go back to the dashboard
- new form, Patient Information, Next, select the (disabled) patient, Esc to close the dialog
- new form: Patient Information is disabled, and selecting any of the other questionnaires still won't let the Continue button work

Probably unrelated, but another bug allows creating multiple Patient Information forms for the same patient:
- lfs mode
- create a new Patient Information and a new Patient (make sure there's only one Patient subject in the system, not more)
- save, go back to the dashboard
- new form, Patient Information, Continue; at this point the only existing patient should be disabled, but also selected; press Create Form -> form is created
- if, however, the only disabled patient is clicked in the previous step, the Create button is correctly disabled

sdumitriu commented:
Oh, the second bug is related, it seems to be exactly what 1206 is reporting. So the issue isn't fixed.

sashaandjic commented:
Confirming bug that @sdumitriu reported.

IntegralProgrammer committed:
LFS-1206: Questionnaire/Subject selector does not reset internal state when closed before completion

Reset the dialog state, regardless of if it is a New Subject Dialog or Questionnaire / Subject selector dialog that is being closed

Additionally, set the disableProgress state variable to false upon dialog state reset

IntegralProgrammer committed:
LFS-1206: Questionnaire/Subject selector does not reset internal state when closed before completion

In SubjectSelectorList, do not show the MaterialTable until the relatedSubjects are loaded

IntegralProgrammer committed:
LFS-1206: Questionnaire/Subject selector does not reset internal state when closed before completion

Eliminate redundantly repeated fetch() with useEffect()

sashaandjic commented:
I have this small complaint, that is probably not worth fixing as it happens only if there is one patient in the system ( and that one is disabled).
- lfs mode
- create a new Patient Information and a new Patient (make sure there's only one Patient subject in the system, not more)
- save, go back to the dashboard
- new form, Patient Information, Continue; at this point the only existing patient should be disabled, **but also selected**

If more than one unavailable patient exist, no patient is selected.

![image](https://user-images.githubusercontent.com/23243244/123345958-33882a00-d525-11eb-907a-15e325a073cb.png)

sashaandjic commented:
REGRESSION:
- create few patients
- go to Forms
- click on `+`
- select Patient Information form
- selection dialog appears 
![image](https://user-images.githubusercontent.com/23243244/123352840-2d4a7b80-d52e-11eb-9170-12ebc69cb8f2.png)
- click on `NEW SUBJECT`
- click on `CANCE:`
- expected behaviour: the selection form from the above is presented
- actual behaviour: this weird dialog appears
![image](https://user-images.githubusercontent.com/23243244/123352971-73074400-d52e-11eb-800e-ff11eb700818.png)
- click on `NEW SUBJECT` again
- this dialog appears 
![image](https://user-images.githubusercontent.com/23243244/123353015-8f0ae580-d52e-11eb-8817-6aa677de22cb.png)
- type the name of the new patient and click on `CREATE`
- a white screen of death appears with a console error greetings
![image](https://user-images.githubusercontent.com/23243244/123353177-f032b900-d52e-11eb-9750-9b720ac709a9.png)
